haskell-src/Concordium/Crypto/EncryptedTransfers.hs

```diff
@@ -297,6 +297,9 @@ instance FromJSON EncryptedAmountTransferData where
         eatdProof <- v .: "proof"
         return EncryptedAmountTransferData{..}
 
+instance ToJSON EncryptedAmountTransferData where
+    toJSON = error "Will not be implemented since feature will be deprecated soon"
```

Comment:
We don't want to update this text again, so I think we should just say that it is already deprecated.

Reply:
I really think we shouldn't have instances like this (i.e. that simply error). `error` should be used with care, and only in places where we can be pretty sure it won't happen, but if it does it's OK to crash.

```suggestion
    toJSON EncryptedAmountTransferData{..} =
        object
            [ "remainingAmount" .= eatdRemainingAmount,
              "transferAmount" .= eatdTransferAmount,
              "index" .= eatdIndex,
              "proof" .= eatdProof
            ]
```

Reply:
we don't want to spend time on implementing this feature since the decision was made to deprecate (even remove) the whole code related to shielded transactions. In that case, I think it is ok to throw an `error` but coping the `toJSON` and not spending time on testing is also fine with me.

Reply:
we don't want to spend time on implementing this feature since the decision was made to deprecate (even remove) the whole code related to shielded transactions. In that case, I think it is ok to throw an error but coping the toJSON and not spending time on testing is also fine with me.